### ISSUE TITLE
runtime error while trying to resurrect fossilized chunks

### ISSUE THREAD

**Issue by Ossssip** [gilbertchen/duplicacy#473]
Hi, I have encountered `runtime error: invalid memory address or nil pointer dereference` while checking a backup. I have managed to find a culprit: it is a bunch of fossilized chunks for one revision of a particular snapshot.

If I run ` duplicacy check -storage mystorage -id mysnapshot -r 41`, I get
```
Storage set to X:\data\
Listing all chunks
Chunk 55438bd93f7d9afd9aa92a58be2228400525a5513bbd810615718e0e568cb0fd referenced by snapshot mysnapshot at revision 41 does not exist
Chunk d27d76a4851b8c6bef57aea840a45e43f32ef948e505c48aadb7884a1d186ba6 referenced by snapshot mysnapshot at revision 41 does not exist

<more chunks missing>

Some chunks referenced by snapshot mysnapshot at revision 41 are missing
```

If run the same command with `-fossils`, I get
```
Storage set to X:\data\
Listing all chunks
Chunk d27d76a4851b8c6bef57aea840a45e43f32ef948e505c48aadb7884a1d186ba6 referenced by snapshot mysnapshot at revision 41 has been marked as a fossil
Chunk 25ca65ded3a8e05f3163d13426ab68034ee3baf2dcfb9588fe9adeea79f3b235 referenced by snapshot mysnapshot at revision 41 has been marked as a fossil

<more chunks are marked as fossils>

All chunks referenced by snapshot mysnapshot at revision 41 exist
```

However, If I try to resurrect them (`duplicacy check -storage mystorage -id mysnapshot -r 41 -fossils -resurrect`), I get

```
Storage set to X:\data\
Listing all chunks
runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0x41, 0x0, 0x0)
        C:/Go/src/runtime/debug/stack.go:24 +0xae
runtime/debug.PrintStack()
        C:/Go/src/runtime/debug/stack.go:16 +0x29
github.com/gilbertchen/duplicacy/src.CatchLogException()
        C:/Users/User/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_log.go:183 +0x1b4
panic(0xd64d40, 0x13e0710)
        C:/Go/src/runtime/panic.go:491 +0x291
github.com/gilbertchen/duplicacy/src.(*ChunkOperator).AddTask(...)
        C:/Users/User/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_chunkoperator.go:94
github.com/gilbertchen/duplicacy/src.(*ChunkOperator).Resurrect(...)
        C:/Users/User/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_chunkoperator.go:111
github.com/gilbertchen/duplicacy/src.(*SnapshotManager).resurrectChunk(0xc0422326c0, 0xc044672690, 0x4c, 0xc0441a1400, 0x40, 0xc044672690)
        C:/Users/User/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_snapshotmanager.go:1520 +0x105
github.com/gilbertchen/duplicacy/src.(*SnapshotManager).CheckSnapshots(0xc0422326c0, 0xc042008130, 0x8, 0xc042234470, 0x1, 0x1, 0x0, 0x0, 0x101000000, 0x0)
        C:/Users/User/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_snapshotmanager.go:850 +0x1041
main.checkSnapshots(0xc042188c60)
        C:/Users/User/go/src/github.com/gilbertchen/duplicacy/duplicacy/duplicacy_main.go:913 +0x5a9
github.com/gilbertchen/cli.Command.Run(0xe8ee04, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0xea9f89, 0x20, 0x0, ...)
        C:/Users/User/go/src/github.com/gilbertchen/cli/command.go:160 +0x82e
github.com/gilbertchen/cli.(*App).Run(0xc042188a20, 0xc04206e000, 0xa, 0x10, 0x0, 0x0)
        C:/Users/User/go/src/github.com/gilbertchen/cli/app.go:179 +0x8c8
main.main()
        C:/Users/User/go/src/github.com/gilbertchen/duplicacy/duplicacy/duplicacy_main.go:2002 +0x590b
```

I use duplicacy freshly built from the source code.

**Comment by Ossssip** [comment]
Meanwhile I have found that I still can resurrect the chunks by running
`duplicacy check -storage mystorage -id mysnapshot -fossils -r 41 -files`:
```
Storage set to X:\data\
Listing all chunks
Fossil cc34cafcde6799fb15a056d4f522c19811de07ccb8642bcf5b5827f6b66d6620 has been resurrected
Fossil 2f159df9509000a18e5645b7e5d7a5bf9a6a317902a9a5f4849e60eae234177f has been resurrected

etc.
```

**Comment by Ossssip** [comment]
Apparently fixed by  f304b64.
